src/recordlinker/database/__init__.py

```diff
@@ -25,10 +27,60 @@ def create_sessionmaker(init_tables: bool = True) -> orm.sessionmaker:
     if settings.connection_pool_max_overflow is not None:
         kwargs["max_overflow"] = settings.connection_pool_max_overflow
     engine = create_engine(settings.db_uri, **kwargs)
+    
     if init_tables:
         models.Base.metadata.create_all(engine)
+    if verify_tables:
+        verify_tables_match_orm(engine)
+
     return orm.sessionmaker(bind=engine)
 
+def are_types_equivalent(db_type, orm_type) -> bool:
+    # Using a str wrapper to normalize types from different contexts (inspector types vs ORM types)
+    db_type_str = str(db_type).lower()
+    orm_type_str = str(orm_type).lower()
+
+    # Treat these cases as equivalent
+    if {db_type_str, orm_type_str} <= {"integer", "bigint"}:
+        return True
+    
+    if {db_type_str, orm_type_str} <= {"double precision", "float"}:
+        return True
+
+    return db_type_str == orm_type_str
+
+def verify_tables_match_orm(engine):
+    """
+    Verify that database tables match ORM definitions.
+    """
+    inspector = inspect(engine)
+    orm_metadata = models.Base.metadata
+
+    for table_name, orm_table in orm_metadata.tables.items():
+        if not inspector.has_table(table_name):
+            raise SQLAlchemyError(
+                f"Table '{table_name}' is missing in the database."
+            )
+
+        db_columns = inspector.get_columns(table_name)
+        db_column_details = {col['name']: col for col in db_columns}
+
+        for orm_column in orm_table.columns:
+            column_name = orm_column.name
+
+            if column_name not in db_column_details:
+                raise SQLAlchemyError(
+                    f"Column '{column_name}' is missing in the database for table '{table_name}'."
+                )
+
+            db_col_type = db_column_details[column_name]['type']
+            orm_col_type = orm_column.type
+
+            if not are_types_equivalent(db_col_type, orm_col_type):
```

Comment:
@cbrinson-rise8 if we explicit compare based on db dialect, does that remove the need for maintaining type mappings?
```python
            if db_col_type.compile(dialect=engine.dialect) != orm_col_type.compile(dialect=engine.dialect):
```